docusaurus/docs/configs/relayminer_config.md

```diff
@@ -0,0 +1,332 @@
+
+The `headers` section of the supplier configuration is a set of key-value pairs
+that will be added to the request headers when the `RelayMiner` forwards the
+requests to the service. It can be used to add additional headers like
+`Authorization: Bearer <TOKEN>` for example.
+
+## `hosts`
+_`Required`_, _`Unique` for each referenced proxy_, _`Unique` within the supplier's `hosts` list_
+
+The `hosts` section of the supplier configuration is a list of hosts that the
+`RelayMiner` will accept requests from. It must be a valid host that reflects
+the on-chain supplier staking service endpoints.
+
+It is used to determine if the incoming request is allowed to be processed by
+the referenced proxy server as well as to check if the request's RPC-Type matches
+the on-chain endpoint's RPC-Type.
+
+The reasons to have multiple hosts for the same supplier service are:
+- The on-chain Supplier may provide the same Service on multiple domains
+(e.g. for different regions).
+- The operator may want to route requests of different RPC types to
+the same proxy
+- Migrating from one domain to another. Where the operator could still
+accept requests on the old domain while the new domain is being propagated.
+
+It must be unique across all the `hosts` lined to a given proxy.
+
+_Note: The `name` of the supplier is automatically added to the `hosts` list as
+it may help troubleshooting the `RelayMiner` and/or send requests internally
+from a k8s cluster for example._
+
+## `proxy_names`
+_`Required`_, _`Unique` within the `proxy_names` list_
+
+The `proxy_names` section of the supplier configuration is the list of proxies
+that the `RelayMiner` will use to serve the requests for the given supplier entry.
+
+It must be a valid proxy name that is defined in the `proxies` section of the
+configuration file, must be unique across the supplier's `proxy_names`  and the
+`supplier` `type` must match the `type` of the referenced `proxy`.
+
+# Proxy -> Supplier referencing
+
+To illustrate how the `proxy_names` and `name` fields are used to reference
+proxies and suppliers, let's consider the following configuration file:
+
+```yaml
+proxies:
+  - name: http-example
+    ...
+  - name: http-example-2
+suppliers:
+  - name: ethereum
+    ...
+    proxy_names:
+      - http-example
+      - http-example-2
+  - name: 7b-llm-model
+    ...
+    proxy_names:
+      - http-example
+```
+
+In this example, the `ethereum` supplier is referencing two proxies, `http-example`
+and `http-example-2` and the `7b-llm-model` supplier is referencing only the
+`http-example` proxy. This would result in the following setup:
+
+```
+- http-example
+  - ethereum
+  - 7b-llm-model
+- http-example-2
+  - ethereum
+```
+
+## RelayMiner config -> On-chain service relationship
+
+The following diagram illustrates how the `RelayMiner` config must match the
+on-chain staked service endpoints.
+
+_Assuming that the on-chain endpoints 1 and 2 have different hosts_
+
+```mermaid
+flowchart LR
```

Comment:
I started making a few edits to the chart so the `yaml <-> protobuf` conversion is clearer. Might need a few more tweaks but lmk wyt


```mermaid
flowchart LR

    subgraph "Supplier Actor (On-Chain)"
        subgraph "SupplierServiceConfig (protobuf)"
            subgraph svc1["Service1 (protobuf)"]
                svc1Id[Service1.Id]
                subgraph SupplierEndpoint
                    EP1[Endpoint1]
                    EP2[Endpoint2]
                end
            end
            subgraph svc2[Service2]
                svc2Id[Service2.Id]
            end
        end
    end

    subgraph "RelayMiner Operator (Off-Chain)"
        subgraph "DevOps Operator Configs (yaml)"
            subgraph svc1Config["Service1 Config (yaml)"]
                svc1IdConfig[name=Service1.Id]-->svc1Id
            subgraph Hosts
                H1[Endpoint1.Host]-->EP1
                H2[Endpoint2.Host]-->EP2
            end
        end
        subgraph svc2Config[Service2 Config]
            svc2IdConfig[Service2.Id]
        end
    end
  end

  svc2Config-->svc2
  ```